src/components/UIUC-Components/MakeNewCoursePage.tsx

```diff
@@ -63,26 +63,21 @@ const MakeNewCoursePage = ({
 
         if (response.ok) {
           const data = await response.json()
-          return data.all_course_names
+          setAllExistingCourseNames(data.all_course_names)
         } else {
           console.error(`Error fetching course metadata: ${response.status}`)
-          return null
         }
       }
 
-      fetchGetAllCourseNames()
-        .then((result) => {
-          setAllExistingCourseNames(result)
-        })
-        .catch((error) => {
-          console.error(error)
-        })
+      fetchGetAllCourseNames().catch((error) => {
```

Comment:
wait this line doesn't do anything, it should be deleted because it makes a whole extra DB request.